src/bridges/tokemak/TokemakBridge.sol

```diff
@@ -0,0 +1,353 @@
+     */
+    function finalise(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 _interactionNonce,
+        uint64
+    )
+        external
+        payable
+        override(BridgeBase)
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool interactionCompleted
+        )
+    {
+        if (msg.sender != ROLLUP_PROCESSOR) revert ErrorLib.InvalidCaller();
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidInputA();
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidOutputA();
+
+        address _tAsset = _inputAssetA.erc20Address;
+        if (_tAsset == address(0)) revert ErrorLib.InvalidInputA();
+
+        // Pending withdrawal value
+        uint256 _inputValue = pendingInteractions[_interactionNonce].inputValue;
+        if (_inputValue == 0) revert ErrorLib.InvalidInput();
+
+        // Withdraw pending withdrawals
+        (outputValueA, interactionCompleted) = _finaliseWithdraw(_tAsset, _inputValue, _interactionNonce);
+    }
+
+    /**
+     * @dev Function called to automatically fetch all the Pools from Tokemak Manager
+     */
+    function loadTokens() public {
+        IManager _manager = IManager(MANAGER);
+        address[] memory _pools = _manager.getPools();
+        for (uint256 i = 0; i < _pools.length; i++) {
+            address _pool = _pools[i];
+            Ttoken _token = Ttoken(_pool);
+
+            address _asset = _token.underlyer();
+            IERC20 _assetToken = IERC20(_asset);
+
+            _assetToken.approve(ROLLUP_PROCESSOR, MAX_UINT);
+            _assetToken.approve(_pool, MAX_UINT);
+            _token.approve(ROLLUP_PROCESSOR, MAX_UINT);
+            _token.approve(_pool, MAX_UINT);
+
+            tTokens[_asset] = _pool;
+            assets[_pool] = _asset;
+        }
+    }
+
+    /**
+     * @dev Function to attempt finalising of as many interactions as possible within the specified gas limit
+     * Continue checking for and finalising interactions until we expend the available gas
+     * @param _gasFloor The amount of gas that needs to remain after this call has completed
+     */
+    function _finalisePendingInteractions(uint256 _gasFloor) internal {
+        // check and finalise interactions until we don't have enough gas left to reliably update our state without risk of reverting the entire transaction
+        // gas left must be enough for check for next expiry, finalise and leave this function without breaching _gasFloor
+        uint256 _gasLoopCondition = MIN_GAS_FOR_CHECK_AND_FINALISE + MIN_GAS_FOR_FUNCTION_COMPLETION + _gasFloor;
+        uint256 _ourGasFloor = MIN_GAS_FOR_FUNCTION_COMPLETION + _gasFloor;
```

Comment:
`_ourGasFloor` is not explanatory name.

Reply:
What do you think is the best name for this?